src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs

```diff
@@ -487,17 +483,31 @@ protected void EnsureStorage()
         /// </summary>
         protected bool IsBoundary(int index, int startpos, int endpos)
         {
-            return (index > startpos && RegexCharClass.IsBoundaryWordChar(runtext![index - 1])) !=
-                   (index < endpos && RegexCharClass.IsBoundaryWordChar(runtext![index]));
+            Debug.Assert(runtext != null, "runtext should not be null since this method is only callable by old codegen.");
+            return (index > startpos && RegexCharClass.IsBoundaryWordChar(runtext[index - 1])) !=
+                   (index < endpos && RegexCharClass.IsBoundaryWordChar(runtext[index]));
+        }
+
+        protected bool IsBoundary(ReadOnlySpan<char> inputSpan, int index, int startpos, int endpos)
```

Comment:
Do we need endpos?  Isn't endpos always inputSpan.Length?

For that matter, isn't startpos always actually runtextbeg?  In which case we don't need it either.  At which point this just takes the span and the index.

Reply:
startpos is not always runtextbeg, but it is runner.runtextstart. I just copied the signature from the existing methods, which they did take start and end but I do think we can just use the regexRunner fields so these will get refactored anyway.

Reply:
> startpos is not always runtextbeg, but it is runner.runtextstart

Where is it not runtextbeg?  I see these two calls in the interpreter:
https://github.com/dotnet/runtime/blob/af42c8ee4e1f0e420605f928cded476520296ced/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs#L713-L727